MainRepository/Program.cs

```diff
@@ -1,7 +1,214 @@
+                  if (j == 0) j--;
+                  else {
+                     mWordle[i, j - 1] = mCircle;
+                     mWordle[i, j] = mDot;
+                     j -= 2;
+                     mList.RemoveAt (mList.Count - 1);
+                  }
+               } else if (char.ToUpper (key.KeyChar) is >= 'A' and <= 'Z') {
+                  if (j < 4) mWordle[i, j + 1] = mCircle;
+                  char c = Convert.ToChar (key.KeyChar);
+                  mWordle[i, j] = char.ToUpper (c);
+                  if (mList.Count < 5) mList.Add (c);
+               }
+               if (j == 4) {
+                  DisplayBoard (word);
+                  while (true) {
+                     var lastKey = Console.ReadKey (true);
+                     if (lastKey.Key == ConsoleKey.Enter && ValidWord ()) {
+                        Console.WriteLine (new string (' ', Console.WindowWidth));
+                        break;
+                     }
+                     if (lastKey.Key == ConsoleKey.Enter && !ValidWord ()) {
+                        mBadWord = string.Join ("", mList).ToUpper ();
+                        var cursorTop = mInvalid ? Console.CursorTop - 1 : Console.CursorTop;
+                        Console.SetCursorPosition (mHalfWidth - 1, cursorTop);
+                        Console.ForegroundColor = ConsoleColor.Yellow;
+                        Console.WriteLine ($"{mBadWord} is not a word");
+                        Console.ResetColor ();
+                        mInvalid = true;
+                     }
+                     if (lastKey.Key == ConsoleKey.Backspace) {
+                        mWordle[i, 4] = mCircle;
+                        mWordle[i + 1, 0] = mDot;
+                        j--;
+                        mList.RemoveAt (4);
+                        Console.SetCursorPosition (mHalfWidth - 1, Console.CursorTop - 1);
+                        mInvalid = false;
+                        Console.WriteLine (new string (' ', Console.WindowWidth));
+                        break;
+                     }
+                  }
+               } else if (key.Key is ConsoleKey.Enter && j < 4) {
+                  j--;
+                  continue;
+               }
+               if (i < 5 && j == 4) mWordle[i + 1, 0] = mCircle;
+               DisplayBoard (word);
+               if (mList.Count == 5) {
+                  DisplayBoard (word, true);
+                  if (mGreenList.Count == 5) {
+                     mGameOver = true;
+                     mCount++;
+                     return;
+                  }
+                  if (!ValidWord ()) j--;
+                  else mList.Clear ();
+               }
+            }
+         }
+         mCount++;
+      }
+   }
+
+   /// <summary>Returns true if the given word is valid</summary>
+   public bool ValidWord () {
+      string input = string.Join ("", mList).ToUpper ();
+      string[] validWords = File.ReadAllLines (@"c:\etc\dict-5.txt");

```

Comment:
Everytime you call ValidWord, this is reading from the file dict-5.txt and checks if the word is valid.
This is not efficient. You should store this in a member variable and access it.
SelectWord is called once so it is fine. But this is called frequently.